padl/transforms.py

```diff
@@ -1461,32 +1507,87 @@ def _pd_evaluable_repr_inner(self, indent: int = 0) -> str:
         return f'~{self.transform._pd_evaluable_repr(indent)}'
 
 
+class Node:
+    """Node wrapper for transforms
+
+    Nodes wrap a transform and can be connected in graph to form complex
+    graphs.
+    :param transform: Transform to wrap
+    """
+    _id = 0
+
+    def __init__(self, transform: Transform):
+        self.id = self._generate_id()  # keep?
+        self.transform = transform
+        self.pd_input_slice = self.transform._pd_input_slice
+        self.pd_output_slice = self.transform._pd_output_slice
+
+    @property
+    def name(self):
+        if self.transform.pd_name is None and isinstance(self.transform, Pipeline):
+            return self.transform._name
+        return self.transform.pd_name
+
+    @property
+    def name_id(self):
+        if self.name is not None:
+            return self.name + ' ' + str(self.id)
+        if isinstance(self.transform, Batchify):
+            name = 'batchify '
+            return name+str(self.id)
+        if isinstance(self.transform, Unbatchify):
+            name = 'unbatchify '
+            return name+str(self.id)
+        return str(self.id)
+
+    @classmethod
+    def _generate_id(cls):
+        cls._id += 1
+        return cls._id
+
+    def __call__(self, args):  # should a node be callable?
+        """Call Method for Node"""
+        return self.call_node(args)
+
+    def copy(self):
+        _copy = type(self)(self.transform)
+        return _copy
```

Comment:
Out of curiosity, when returning a copy we do want to keep the same `Node.id` or should it be incremented?

Reply:
good point!

Reply:
I think it should be incremented, as a copy is new instance and is expected to be a new instance.

Reply:
Okay that sounds good to me!